src/model/CoilCoolingDX.cpp

```diff
@@ -144,11 +156,70 @@ namespace model {
         }
       }
 
+      // CoilSystemCoolingDXHeatExchangerAssisted
+      {
+        auto coilSystems = this->model().getConcreteModelObjects<CoilSystemCoolingDXHeatExchangerAssisted>();
+        for (const auto& coilSystem : coilSystems) {
+          if (coilSystem.coolingCoil().handle() == this->handle()) {
+            return coilSystem;
+          }
+        }
+      }
+
+      return boost::none;
+    }
+
+    boost::optional<ZoneHVACComponent> CoilCoolingDX_Impl::containingZoneHVACComponent() const {
```

Comment:
Add `containingZoneHVACComponent` method as we can now have `Coil:Cooling:DX` on both `ZoneHVAC:PackagedTerminalAirConditioner` and `ZoneHVAC:PackagedTerminalHeatPump`.